### PR TITLE
fix py27 compat

### LINKED ISSUE #2547: py=2, py<3.5 compatibility
https://github.com/iterative/dvc/blob/5c76b46b460e00d3bb5c3f2ea87b19d961491f12/dvc/logger.py#L161

is invalid for `python<3.5`.

### PR DESCRIPTION
**Pull request by casperdcl**
- fix #2547 using same approach as https://github.com/matplotlib/matplotlib/pull/10422

### CODE REVIEW

**Comment by casperdcl** [comment]
not sure how/if to make deepSource happy

File: dvc/logger.py [hunk 1]
```diff
@@ -10,6 +10,11 @@
 import logging.config
 import colorama
 
+try:  # https://docs.python.org/3/library/exceptions.html#RecursionError
+    RecursionError
+except NameError:  # Python < 3.5
+    RecursionError = RuntimeError
```
Comment (efiop) [inline dvc/logger.py]: How about we move it to `dvc/utils/compat.py`, so it is easier to get rid of it next year when py35 is dead?

**Comment by efiop** [comment]
> not sure how/if to make deepSource happy

it was complaining about RecursionError redefined, which was intentional, so I've : simply approved it.

File: dvc/utils/compat.py [hunk 2]
```diff
@@ -18,6 +18,11 @@
 # simplified version of ipython_genutils/encoding.py
 DEFAULT_ENCODING = sys.getdefaultencoding()
 
+if _ver[:2] < (3, 5):
+    RecursionError = RuntimeError
```
Comment (casperdcl) [inline dvc/utils/compat.py]: really don't like this but it was the simplest way I could expose this for external

**Pull request merged**